lib/elixir/lib/option_parser.ex

```diff
@@ -114,6 +117,31 @@ defmodule OptionParser do
   end
 
   @doc """
+  The same as `parse/2` but raises an `ArgumentError` if it fails.
+  Returns the parsed values, remaining arguments and an empty list with the
+  invalid options otherwise.
```

Comment:
When I read the docstring for the first time, I tripped on the phrase "empty list with the invalid options". I think it can be made clearer:

> The same as `parse/2` but raises an `ArgumentError` if any invalid options are encountered.
>
> If there weren't any errors, returns the parsed options, remaining arguments and an empty list.

Also, in the docstring for `parse/2` we should likewise change `Returns the parsed values` to `Returns the parsed options`.